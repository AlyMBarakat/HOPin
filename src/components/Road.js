import React, {useState, useEffect} from 'react';
import {
  View,
  Text,
  StyleSheet,
  TouchableOpacity,
  LayoutAnimation,
} from 'react-native';

import RideCard from './RideCard';

const temp = [
  {
    name: 'Aly',
    car: 'Mitsubishi Lancer 2007',
  },
  {
    name: 'Khalid Khalil',
    car: 'Fiat Tipo',
  },
  {
    name: 'Amira Barakat',
    car: 'Nissan Qashqai',
  },
];

const Road = ({source = 'Cairo', destination = 'Alexandria'}) => {
  const [ridesShown, setRides] = useState(false);

  return (
    <View>
      <TouchableOpacity
        style={styles.container}
        onPress={() => {
          // animate components transitions
          LayoutAnimation.configureNext(LayoutAnimation.Presets.easeInEaseOut);
          setRides(ridesShown => !ridesShown);
        }}>
        <Text style={{fontSize: 20, color: '#fff', fontWeight: 'bold'}}>
          {source} {'->'} {destination}
        </Text>
      </TouchableOpacity>
      {ridesShown &&
        temp.map((item, index) => {
          return <RideCard key={index} name={item.name} car={item.car} />;
        })}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#1F2020',
    padding: 17,
    borderRadius: 5,
    marginBottom: 10,
    flexDirection: 'row',
    justifyContent: 'center',
  },
});

export default Road;
