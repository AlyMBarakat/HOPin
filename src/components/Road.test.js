import React from 'react';
import renderer from 'react-test-renderer';
import Road from './Road';

const temp = [
  {
    name: 'Aly',
    car: 'Mitsubishi Lancer 2007',
  },
  {
    name: 'Khalid Khalil',
    car: 'Fiat Tipo',
  },
  {
    name: 'Amira Barakat',
    car: 'Nissan Qashqai',
  },
];

const testList = ['temp1', 'temp2', 'temp3', 'temp4', 'temp5', 'temp6'];

test('renders date and time pickers correctly', () => {
  const tree = renderer
    .create(<Road source="Alexandria" destination="Cairo" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
