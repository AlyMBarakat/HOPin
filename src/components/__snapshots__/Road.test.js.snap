// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders date and time pickers correctly 1`] = `
<View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#1F2020",
        "borderRadius": 5,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 10,
        "opacity": 1,
        "padding": 17,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      Alexandria
       
      -&gt;
       
      Cairo
    </Text>
  </View>
</View>
`;
